### ISSUE TITLE
Exchangeable backends

### ISSUE THREAD

**Issue by nightscape** [burhanrashid52/WhatTodo#30]
Hi @burhanrashid52,

I'm currently in the process of switching from Todoist to org-mode as project management solution.
One thing where org-mode is a bit behind is a good-looking UI for Android.
There's [orgzly](https://github.com/orgzly/orgzly-android) and it does its job, but it's far from being intuitive.
Now I'm wondering if it would be possible to use WhatTodo as a frontend to org-mode.
In order to do that, one would have to hide the current `*Database` classes behind `abstract class`es and provide another implementation of those classes using org-mode (potentially based on code from [orgro](https://github.com/amake/orgro)).

I'm not a 100% how much effort this would be and if I want to invest the effort, but I wanted to ask in advance if this is a direction you could see the project evolving to, i.e. WhatTodo being really nice UI for different "storage mechanisms", with a built-in DB being the default.

**Comment by nightscape** [comment]
Another interesting option might be to use something like [Firebase Cloud Firestore](https://firebase.flutter.dev/docs/firestore/usage/) to turn this app into a collaborative one, which would almost make it a replacement for Todoist.

**Comment by burhanrashid52** [comment]
@nightscape Yes. I agree with you and I am also looking forward in the same direction. Instead of a concrete class, it should expose the abstract class. This will allow us to swipe the data source. Ex Firebase.
A good starting point to learn this is from [flutter_architecture_samples](https://github.com/brianegan/flutter_architecture_samples) where they have different implementation of `TodoRepository`.

**Comment by burhanrashid52** [comment]
I will be happy to see a PR on this :)
Let me know if you want to discuss more. We can move this into GitHub discussion.

**Comment by nightscape** [comment]
The flutter_architecture_samples are exactly what I had in mind!
Down to the point where they have a `ReactiveTodosRepository` which streams the current Todos and therefore allows for real-time updates.
